sdk/storage/storage-blob/src/generated/src/storageClient.ts

```diff
@@ -6,32 +6,74 @@
  * Changes may cause incorrect behavior and will be lost if the code is regenerated.
  */
 
+import * as coreHttpCompat from "@azure/core-http-compat";
+import {
+  ServiceImpl,
+  ContainerImpl,
+  BlobImpl,
+  PageBlobImpl,
+  AppendBlobImpl,
+  BlockBlobImpl
+} from "./operations";
 import {
   Service,
   Container,
   Blob,
   PageBlob,
   AppendBlob,
   BlockBlob
-} from "./operations";
-import { StorageClientContext } from "./storageClientContext";
+} from "./operationsInterfaces";
 import { StorageClientOptionalParams } from "./models";
 
-export class StorageClient extends StorageClientContext {
+export class StorageClient extends coreHttpCompat.ExtendedServiceClient {
+  url: string;
+  version: string;
+
   /**
    * Initializes a new instance of the StorageClient class.
    * @param url The URL of the service account, container, or blob that is the target of the desired
    *            operation.
    * @param options The parameter options
    */
   constructor(url: string, options?: StorageClientOptionalParams) {
-    super(url, options);
-    this.service = new Service(this);
-    this.container = new Container(this);
-    this.blob = new Blob(this);
-    this.pageBlob = new PageBlob(this);
-    this.appendBlob = new AppendBlob(this);
-    this.blockBlob = new BlockBlob(this);
+    if (url === undefined) {
+      throw new Error("'url' cannot be null");
+    }
+
+    // Initializing default values for options
+    if (!options) {
+      options = {};
+    }
+    const defaults: StorageClientOptionalParams = {
+      requestContentType: "application/json; charset=utf-8"
+    };
+
+    const packageDetails = `azsdk-js-azure-storage-blob/12.12.1`;
+    const userAgentPrefix =
```

Comment:
In pipeline.ts, similar thing is also done for `userAgentPrefix`.  Do we need that if it is already handled here.

Reply:
Yeah, since Storage manually creates its own custom pipeline this is not needed, but this file is autogenerated to work without custom pipelines